.github/workflows/build.yml

```diff
@@ -26,10 +27,10 @@ jobs:
 
     - name: Install Pulumi CLI
       uses: pulumi/setup-pulumi@v2
-    - name: Set up Python 3.11
+    - name: Set up Python 3.12
```

Comment:
Why was this needed?

Reply:
I use some features in the tests which are only available in 3.12